File: src/main/java/com/travel/domain/city/dto/CityDto.java
package com.travel.domain.city.dto;

import com.travel.domain.city.domain.City;
import com.travel.domain.city.domain.CityCategory;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class CityDto {
    private Long id;
    private String name;
    private CityCategory category;

    @Builder
    public CityDto(Long id, String name, CityCategory category) {
        this.id = id;
        this.name = name;
        this.category = category;
    }

    public static CityDto of(City city) {
        return new CityDto(
                city.getId(),
                city.getName(),
                city.getCategory()
        );
    }
}

File: src/main/kotlin/com/travel/domain/city/dto/CityDto.kt
package com.travel.domain.city.dto

import com.travel.domain.city.domain.City
import com.travel.domain.city.domain.CityCategory

data class CityDto(
        val id: Long,
        val name: String,
        val category: CityCategory) {
    companion object {
        fun of(city: City): CityDto {
            return CityDto(
                    id = city.id!!,
                    name = city.name,
                    category = city.category
            )
        }
    }
}